FakeItEasy.Deploy/CommandLineOptions.cs

```diff
@@ -0,0 +1,83 @@
+namespace FakeItEasy.Deploy;
+
+internal class CommandLineOptions
+{
+    public string Repo { get; private set; } = string.Empty;
+    public string TagName { get; private set; } = string.Empty;
+    public string ArtifactsFolder { get; private set; } = string.Empty;
+
+    public static CommandLineOptions Parse(string[] args)
+    {
+        Action<string>? nextArgAction = null;
+        var options = new CommandLineOptions();
+        foreach (var arg in args)
+        {
+            if (nextArgAction is not null)
```

Comment:
This is a clever pattern.

Reply:
I _didn't_ find it too clever. Just enough.

Oh, good catch.

Reply:
Alternative implementation:
```csharp
public static CommandLineOptions Parse2(string[] args)
{
    string repo = string.Empty;
    string tagName = string.Empty;
    string artifactsFolder = string.Empty;
    bool dryRun = false;

    for (int i = 0; i < args.Length; i++)
    {
        switch (args[i])
        {
            case "-r":
            case "--repo":
                repo = ReadNext();
                break;
            case "-t":
            case "--tag-name":
                tagName = ReadNext();
                break;
            case "-a":
            case "--artifacts-folder":
                artifactsFolder = ReadNext();
                break;
            case "-d":
            case "--dry-run":
                dryRun = true;
                break;
            default:
                throw new Exception($"Unknown command line option '{args[i]}'");
        }

        string ReadNext()
        {
            if (i + 1 >= args.Length)
            {
                throw new Exception($"Expected a value for option '{args[i]}', but none was provided");
            }

            return args[++i];
        }
    }
    
    return new CommandLineOptions(repo, tagName, artifactsFolder, dryRun);
}
```
Less clever, and more correct...
What do you think?

Reply:
Yeah. Cut out the clever.